app/controllers/api/conditions_controller.rb

```diff
@@ -1,4 +1,19 @@
 module Api
   class ConditionsController < BaseController
+    def create_resource(type, _id, data = {})
+      assert_id_not_specified(data, type)
+      data["expression"] = MiqExpression.new(data["expression"])
```

Comment:
Should be checking that data["expression"] is specified,   if required should raise a BadRequestError if not specified.